### PR TITLE
move self-test from Bio/PDB/Polypeptide.py

### PR DESCRIPTION
**Pull request by Gasta88**
In reference to issue #820 and old PR #1177 .

### CODE REVIEW

File: Bio/PDB/Polypeptide.py [hunk 1]
```diff
@@ -452,35 +452,3 @@ def _test_dist(self, c, n):
         else:
             return 0
 
-
```
Comment (peterjc) [inline Bio/PDB/Polypeptide.py]: You've left a blank line at the end of the file, which is why the TravisCI style check is failing:


```
$ flake8 Bio/
Bio/PDB/Polypeptide.py:454:1: W391 blank line at end of file
```

**Comment by peterjc** [comment]
The main reason this test fails as written is the floating point numbers do not match exactly. This is to be expected when comparing different operating systems, libraries, compiler optimisations etc. e.g.

```
======================================================================
FAIL: test_polypeptide (test_PDB.PolypeptideTests)
Tests on polypetide class and methods.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_PDB.py", line 1391, in test_polypeptide
    self.assertEqual(pp[0].get_phi_psi_list(), [(None, -0.46297171497725553), (-1.0873937604007962, 2.1337707832637109), (-2.4052232743651882, 2.3807316946081558), (-1.1914372943688751, 2.2161228118006582), (-1.167611340740029, 2.63504626551578), (-0.99335831029928101, -0.40970857682803996), (-2.1102229342150527, 0.17890937328728854), (-1.4642755598371093, 2.2293202325745112), (-1.0140706072511441, 2.4596685907543421), (-0.94947474533884979, -0.73900573394588243), (-1.0520382830532629, -0.66228271877762757), (-1.1604702715237931, -0.71144976594057252), (-1.1433895379832477, -0.73988272191856985), (-1.0116813564889713, -0.76732761159664509), (-1.1245049821816666, -0.73669003771671482), (-1.1188875868971477, -0.77208804536726727), (-0.99944855432402868, -0.8619934702849692), (-1.2218445209967945, -0.41278852848807851), (-1.179752323229885, -0.74792816563896858), (-1.2158962804334972, -0.60538112130585664), (-1.2012963395236214, -0.66151598677994561), (-1.098286327338313, -0.59288000176128175), (-1.0587237360099633, -0.32715161475523286), (-1.8180488653266087, 0.076317023651986796), (0.81096454256933281, 0.8307201820816954), (-1.0858318540132235, 2.4421060464231292), (-1.317788406220783, 3.1054909566220061), (-1.0724955936835108, -0.80881499531395828), (-1.0310715826779675, -0.74074258319072428), (-1.1338479074559662, -0.69806535315597451), (-1.0773262480370305, -0.74961850479413727), (-1.0646491069070618, -0.79609255605099372), (-1.079209857767184, None)])
AssertionError: Lists differ: [(None, -0.46297171497725553),... != [(None, -0.46297171497725553),...
First differing element 1:
(-1.0873937604007962, 2.1337707832637114)
(-1.0873937604007962, 2.133770783263711)
Diff is 4617 characters long. Set self.maxDiff to None to see it.
----------------------------------------------------------------------
```

There are numpy functions for comparing almost equal arrays which would probably be sensible to use here.

**Comment by Gasta88** [comment]
The presence of `None` elements is interfering with numpy functions to compare arrays.

Running some tests locally with plain `==` operator seem to yield positive results.

**Comment by peterjc** [comment]
Hmm. There are probably options in numpy to make an array mapping ``None`` to the special float ``NaN``.... or you could then the array into a string using (say) 2dp and compare that... but maybe it would be simplest to just compare a subset of the values?

e.g. Just check the first three set of angle pairs?

**Comment by Gasta88** [comment]
I'll do a check of the first three elements, but explicitly by taking one element at the time.

I would like to see if the most basic solution raise any error.

**Comment by peterjc** [comment]
I'd suggest calling ``pp[0].get_phi_psi_list()`` once and storing the result, e.g.

```
phi_psi = pp[0].get_phi_psi_list()
```

And then doing the tests on ``phi_psi``. I think this would be clearer, and it should be marginally quicker too.

**Comment by peterjc** [comment]
It looks like you'd best to the same for the tau angles, that's failing due to a similar close but not equal floating point comparison:

```
======================================================================
FAIL: test_polypeptide (test_PDB.PolypeptideTests)
Tests on polypetide class and methods.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biopython/biopython/Tests/test_PDB.py", line 1414, in test_polypeptide
    self.assertEqual(pp[1].get_tau_list(), [0.3597907225123525, 0.43239284636769254, 0.99820157492712114, 0.6404992574459768, 0.86291043834277825, 0.8364364007031726, 1.9208425755291829, -1.6418305724904823, -1.6531461925249447, 0.8152943543297706, 0.62166174974010979, 1.1296800024708877, 0.85274229666614465, 0.84910125014427418, 0.8792136402844658, 0.89561671990228009, 1.0813933217353846, 1.2736678454416308, -2.5364947072163697, -0.50581980381973402, 0.48324880548682114, 2.6915888387001483, -2.2407955990017334, -1.938889842362177, 0.97306623802066594])
AssertionError: Lists differ: [0.3597907225123525, 0.4323928... != [0.3597907225123525, 0.4323928...
First differing element 2:
0.998201574927
0.998201574927
Diff is 1337 characters long. Set self.maxDiff to None to see it.
----------------------------------------------------------------------
```

But otherwise I'm happy with how this pull request is looking 👍

**Comment by Gasta88** [comment]
Thanks Peter!

**Comment by peterjc** [comment]
Lovely, thank you :)

**Pull request merged**